akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -74,9 +75,12 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
     BidiShape(substreamIn, frameOut, frameIn, substreamOut)
 
   def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
-    new GraphStageLogic(shape) with BufferedOutletSupport { logic ⇒
-      case class SubStream(
+    new GraphStageLogic(shape) with BufferedOutletSupport {
+      logic ⇒
+
+      final case class SubStream(
         streamId:                  Int,
+        headers:                   HeadersFrame,
```

Comment:
We might not need to keep it here. There can only be one HEADERS frame ongoing at any point. So, at the point we start reading a `HEADERS` frame with `endHeaders = false` we can just move into a special state where we accept only `CONTINUATION` frames and err-out otherwise.